simexpal/launch/common.py

```diff
@@ -250,16 +265,61 @@ def compile_manifest(run):
 		'timeout': float(exp.info.timeout) if exp.info.timeout is not None else exp.info.timeout,
 		'environ': environ,
 		'output': exp.info.output,
+		'output_extensions': exp.info.output_extensions,
+		'stdout': exp.info.stdout,
 		'workdir': exp.info.workdir
 	})
 
 def invoke_run(manifest):
+
+	def get_qualified_output_file(ext):
+		if ext not in set(manifest.output_extensions + ['out']):
+			raise RuntimeError(
+				f"Unexpected output extension for experiment '{manifest.experiment}': .{ext}\n"
+			)
+		return manifest.output_file_path(ext)
+
+	def substitute_stdout_key(p):
+		if p == 'OUTPUT':
+			return manifest.output_file_path('out')
+		elif p.startswith('OUTPUT:'):
+			return get_qualified_output_file(p.split(':')[1])
+		else:
+			return None
+
+	def stdout_is_valid():
+		for ext in set(manifest.output_extensions + ['out']):
+			if manifest.stdout == manifest.output_file_path(ext):
+				return True
+		return False
+
+	if manifest.output == 'stdout':
+		output_path = manifest.output_file_path('out')
+
+		msg = "Specifying the output path in the 'output' key is deprecated and will be removed in future " \
+				"versions. Use the 'stdout' key instead."
+		warnings.warn(msg, DeprecationWarning)
+	elif manifest.stdout is not None:
+		if manifest.stdout.startswith('@') and manifest.stdout.endswith('@'):
+			output_path = util.expand_at_params(manifest.stdout, substitute_stdout_key)
+		else:
+			if stdout_is_valid():
+				output_path = manifest.stdout
+			else:
+				raise RuntimeError("The path '{}' is not valid for the 'stdout' key of the experiment "
+					"'{}' with variants '{}', revision '{}' and instance '{}'. Use the "
+					"@-variables '@OUTPUT@', '@OUTPUT:<ext>@' or a valid stdout path.".format(
+					manifest.stdout, manifest.experiment, manifest.variants,
+					manifest.revision if manifest.revision is not None else '', manifest.instance))
+	else:
+		output_path = manifest.aux_file_path('stdout')
+
 	# Create the output file. This signals that the run has been started.
 	(stdout_pipe, stdout) = (None, None)
 	with open(manifest.output_file_path('out'), "w") as f:
 		# We do not actually need to write anything to the output file.
 		# However, we might want to pipe experimental output to it.
-		if manifest.output == 'stdout':
+		if output_path == manifest.output_file_path('out'):
```

Comment:
Why is this the right check? We always want to create the `.out` file eagerly (because status checks rely on it, right?). We might also want to create the other output files (= those with other extensions eagerly). The only case where we want to create a file lazily is the stdout file in aux. This file is only used if the user does *not* care about stdout (i.e., `stdout:` is not given).

Reply:
This check should be equivalent to the check that we had before. When ``manifest.output == 'stdout'``, then the output path is ``manifest.output_file_path('out'``). But now we have can achieve this via 2 different keys, namely the old ``'output'`` (with value ``'stdout'``)  and new ``'stdout'`` (with value ``@OUTPUT@`` or ``@OUTPUT:out@``)  key. Thats why I used the new variable ``output_path`` to catch both cases. The ``.out`` file will always we created, no matter the case (due to the ``with open(manifest.output_file_path('out'), 'w') as f:`` line that comes before the marked line)